recipe_batches/recipe_batches.py

```diff
@@ -2,13 +2,31 @@
 
 import math
 
-def recipe_batches(recipe, ingredients):
-  pass 
+def recipe_batches(recipe, ingredients, batches=0):
+  batch = batches
+  pendingBatch = []
+  for ingredient in recipe:
+    if ingredient in ingredients: #check if it has ingredient at all
+      if recipe.get(ingredient) <= ingredients.get(ingredient): #check if it has enough ingredient
+        pendingBatch.append(1) #if so, add a tally to pending
+        ingredients[ingredient] -= recipe.get(ingredient) #decrement from ingredients
+        if len(pendingBatch) == len(recipe): #check if the pending batch is finished
+          batch += 1 #if it is, increment batch
+          pendingBatch = [] #and reset pending
+          recipe_batches(recipe, ingredients, batch) #try again to make another batch
```

Comment:
I think I see why you're getting `None`. In order to get something out of a function, recursively called or not, you need a `return`. In line 16, you are making a recursive call to `recipe_batches()`. Ultimately, your base cases will return either `batch` or `0`, but without a return here, you will never see it. Your base case execution contexts will resolve, then resolve upwards until it hits here, but this execution context will never cough up the output without a return.

TLDR: Put a `return` before the recursive function call on line 16.